### PR TITLE
Graphql changes for new labels schema

### PR DESCRIPTION
**Pull request by kamaleshnneerasa**
## Description
Please include a summary of the change, motivation and context.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: hypertrace-graphql-labels-schema/src/main/java/org/hypertrace/graphql/label/schema/mutation/LabelMutationSchema.java [hunk 1]
```diff
@@ -22,5 +22,5 @@
   @GraphQLNonNull
   @GraphQLName(UPDATE_LABEL)
   @GraphQLDataFetcher(LabelUpdateMutator.class)
-  Label updateLabel(@GraphQLNonNull @GraphQLName(Label.ARGUMENT_NAME) Label label);
+  Label updateLabel(@GraphQLNonNull @GraphQLName(UpdateLabel.ARGUMENT_NAME) UpdateLabel label);
```
Comment (aaron-steinfeld) [inline hypertrace-graphql-labels-schema/src/main/java/org/hypertrace/graphql/label/schema/mutation/LabelMutationSchema.java:25]: nit: match code arg names to declared arg names
```suggestion
  Label updateLabel(@GraphQLNonNull @GraphQLName(UpdateLabel.ARGUMENT_NAME) UpdateLabel updateLabel);
```

(same for create)

File: hypertrace-graphql-labels-schema/src/main/java/org/hypertrace/graphql/label/schema/query/Label.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  String CREATED_BY_KEY = "createdBy";
+
+  @GraphQLField()
+  @GraphQLName(LABEL_DATA_KEY)
+  @GraphQLNonNull()
+  LabelData data();
+
+  @GraphQLField()
+  @GraphQLName(CREATED_BY_RULE_ID_KEY)
+  @GraphQLNonNull
```
Comment (aaron-steinfeld) [inline hypertrace-graphql-labels-schema/src/main/java/org/hypertrace/graphql/label/schema/query/Label.java:24]: This field is optional, so should be nullable

File: hypertrace-graphql-labels-schema/src/main/java/org/hypertrace/graphql/label/schema/mutation/CreateLabel.java [hunk 3]
```diff
@@ -3,15 +3,24 @@
...
   String TYPE_NAME = "CreateLabel";
-  String ARGUMENT_NAME = "label";
-  String KEY = "key";
+  String ARGUMENT_NAME = "createLabel";
 
-  @GraphQLField
-  @GraphQLNonNull
-  @GraphQLName(KEY)
-  String key();
+  String CREATED_BY_RULE_ID_KEY = "createdByRuleId";
```
Comment (aaron-steinfeld) [inline hypertrace-graphql-labels-schema/src/main/java/org/hypertrace/graphql/label/schema/mutation/CreateLabel.java:14]: This is only for internally created rules, need not expose on the API for write

File: hypertrace-graphql-labels-schema/src/main/java/org/hypertrace/graphql/label/dao/LabelResponseConverter.java [hunk 4]
```diff
@@ -5,51 +5,72 @@
...
+  LabelData convertLabelData(org.hypertrace.label.config.service.v1.LabelData data) {
+    return new ConvertedLabelData(data.getKey(), data.getColor(), data.getDescription());
   }
 
   @Value
   @Accessors(fluent = true)
-  private static class DefaultLabelResultSet implements LabelResultSet {
-    static final LabelResultSet EMPTY_LABEL_RESULT_SET = new DefaultLabelResultSet(List.of(), 0, 0);
+  private static class ConvertedLabelResultSet implements LabelResultSet {
+    static final LabelResultSet EMPTY_LABEL_RESULT_SET =
```
Comment (aaron-steinfeld) [inline hypertrace-graphql-labels-schema/src/main/java/org/hypertrace/graphql/label/dao/LabelResponseConverter.java:54]: Is this ever used?

Comment (kamaleshnneerasa) [reply hypertrace-graphql-labels-schema/src/main/java/org/hypertrace/graphql/label/dao/LabelResponseConverter.java:54]: I think it is used in joiner.

**Review by aaron-steinfeld — COMMENTED** [review]
Please revert the submodule change - this PR is rolling back the submodules several months  (from `f50dca8` to `2edb271`)

**Review by aaron-steinfeld — CHANGES_REQUESTED** [review]
Sorry for just picking up on this, but the GQL changes were only meant to be in the GQL<-> config service adapter - that is, the DAO.

Changing the GQL schema here will break the UI, which we can't do.

**Pull request closed without merging**